index.rst

```diff
@@ -17,41 +17,72 @@ Adafruit CircuitPython API Reference
-<https://adafruit.com>`_. Adafruit CircuitPython features unified Python core
-APIs and a growing list of drivers that work with it.
-
-The Adafruit Express line of boards are specifically
-designed for use with CircuitPython. They are the
-`CircuitPlayground Express <https://www.adafruit.com/product/3333>`_,
-`Feather M0 Express <https://www.adafruit.com/product/3403>`_,
-and `Metro M0 Express <https://www.adafruit.com/product/3505>`_.
-Other supported boards include the `Arduino Zero
-<https://www.arduino.cc/en/Main/ArduinoBoardZero>`_, `Adafruit Feather M0 Basic
-<https://www.adafruit.com/product/2772>`_, `Adafruit Feather HUZZAH
-<https://www.adafruit.com/products/2821>`_ and `Adafruit Feather M0 Bluefruit LE
-<https://www.adafruit.com/products/2995>`_.
-
-`Adafruit <https://adafruit.com>`_ has many excellent tutorials available
-through the `Adafruit Learning System <https://learn.adafruit.com/>`_. These
-docs are low-level API docs and may link out to separate getting started guides.
+Welcome to the API reference documentation for Adafruit CircuitPython.
+This contains low-level API reference docs which may link out to separate
+*"getting started"* guides. `Adafruit <https://adafruit.com>`_ has many 
+excellent tutorials available through the
+`Adafruit Learning System <https://learn.adafruit.com/>`_.
+
+What is CircuitPython?
+----------------------
+
+**CircuitPython** is an *education friendly* open source derivative of 
+`MicroPython <https://micropython.org>`_. CircuitPython supports use on
+educational development boards designed and sold by
+`Adafruit <https://adafruit.com>`_. Adafruit CircuitPython features unified
+Python core APIs and a growing list of Adafruit libraries and drivers of
+that work with it.
+
+What boards use CircuitPython?
+------------------------------
+
+The Adafruit Express line of boards is specifically
+designed for use with CircuitPython. This line includes:
+`CircuitPlayground Express <https://www.adafruit.com/product/3333>`_;
+`Feather M0 Express <https://www.adafruit.com/product/3403>`_; and
+`Metro M0 Express <https://www.adafruit.com/product/3505>`_
+
+Other supported boards include:
+`Arduino Zero <https://www.arduino.cc/en/Main/ArduinoBoardZero>`_;
+`Adafruit Feather M0 Basic <https://www.adafruit.com/product/2772>`_;
+`Adafruit Feather HUZZAH <https://www.adafruit.com/products/2821>`_; and
+`Adafruit Feather M0 Bluefruit LE <https://www.adafruit.com/products/2995>`_.
+
+.. _contents:
+
+Full Table of Contents
+----------------------
 
 .. toctree::
-    :maxdepth: 3
-
-    shared-bindings/index.rst
-    docs/supported_ports.rst
-    docs/troubleshooting.rst
-    docs/drivers.rst
-    docs/design_guide
-    docs/common_hal
-    docs/library/index.rst
-    README
-    CONTRIBUTING
-    CODE_OF_CONDUCT
-    license.rst
+   :maxdepth: 3
+   :caption: API and Usage
+
+   shared-bindings/index.rst
+   docs/supported_ports.rst
+   docs/troubleshooting.rst
+   docs/drivers.rst
+
+.. toctree::
```

Comment:
Could we reduce the maxdepth of this tree and the two below? I think its ok to have people dig into it.

Reply:
absolutely, just let me know what you would like it to be? 2? 1?

On the doc build itself, I think I have it down to about 5-7 warnings after a big refactor. Still not completely happy with it as the build of MP is overly complex. Once I get it stable, I'll send you a link to the rendered docs via a WIP PR.

Reply:
I think 1 so just the first level shows.